src/zfm20/zfm20.cxx

```diff
@@ -540,10 +439,10 @@ uint8_t ZFM20::deleteDB()
   return rPkt[9];
 }
 
-uint8_t ZFM20::search(int slot, uint16_t *id, uint16_t *score)
+uint8_t ZFM20::search(int slot, uint16_t &id, uint16_t &score)
 {
```

Comment:
Verified with python using typemaps to handle the references.  IMO, this is easier (at least in python) since it returns a list (example search(1) = [0, 1, 80]) but I will double-check for java and javascript.  The current pointer implementation in C++ is weird and missing any check before de-referencing the pointers.